pkg/controller/operators/olm/operator.go

```diff
@@ -396,7 +396,14 @@ func (a *Operator) handleClusterServiceVersionDeletion(obj interface{}) {
 
 	defer func(csv v1alpha1.ClusterServiceVersion) {
 		logger.Debug("removing csv from queue set")
-		a.csvQueueSet.Remove(csv.GetName(), csv.GetNamespace())
+		if err := a.csvQueueSet.Remove(csv.GetName(), csv.GetNamespace()); err != nil {
+			logger.WithError(err).Debug("error removing from queue")
+		}
+
+		if clusterServiceVersion.IsCopied() {
```

Comment:
This looks good. One structural thing is that we could close over a `var copied *bool` initialized to `false`, set it below when we check `clusterServiceVersion.IsCopied()`, and use that here to avoid checking again (not that it's very resource intensive).